src/App.js

```diff
@@ -67,6 +67,17 @@ class Root extends React.Component {
       // We are still receiving lot of ws messages that are destined to the admin-frontend and not this wallet
       // TODO separate those messages
       if (wsData.type === 'wallet:address_history') {
+        // If is a new transaction, we send a notification to the user, in case it's turned on
+        if (!wallet.txExists(wsData.history)) {
+          const message = 'You\'ve received a new transaction! Click to open it.'
+          const notification = wallet.sendNotification(message);
+          // Set the notification click, in case we have sent one
+          if (notification !== undefined) {
+            notification.onclick = () => {
```

Comment:
Can't we have this part inside `wallet.sendNotification`? We'd just have to also pass the `tx_id` as a parameter. I think it makes more sense to handle everything about notifications there.

Reply:
The idea is that the `wallet.sendNotification` would be generic to send a notification for all the system, that's why I send the `message` already formatted for it.

I could send also a callback to be used onclick but I preferred to do this specific for each type of notification. I don't think the `sendNotification` method should receive a `tx_id`, since other notifications won't need it.

Reply:
Sending the callback is a good idea. But we can do it later, when/if we starting using `wallet.sendNotification ` in more places.

Reply:
Ok